src/HospitalLibrary/BloodStorages/BloodStorageService.cs

```diff
@@ -40,5 +45,72 @@ public async Task<BloodStorage> GetByType(BloodType type)
             BloodStorage blood = await _unitOfWork.BloodStorageRepository.GetByType(type);
             return blood;
         }
+
+        public List<BloodType> GetAllCompatibileBloodStorage(int hospitalizationId)
+        {
+            Hospitalization hospitalization = _unitOfWork.HospitalizationRepository.GetOne(hospitalizationId);
+            MedicalRecord medicalRecord = _unitOfWork.MedicalRecordRepository.GetOne(hospitalization.MedicalRecordId);
+            Patient patient = _unitOfWork.PatientRepository.GetOne(medicalRecord.PatientId);
+            BloodType bType = patient.BloodType;
+            List<BloodType> compatibile = FindCompatibile(bType);
+            return compatibile;
+        }
+
+        private List<BloodType> FindCompatibile(BloodType bType)
+        {
+            List<BloodType> compatibile = new List<BloodType>();
+            if (bType == BloodType.A_NEGATIVE)
+            {
+                compatibile.Add(BloodType.A_NEGATIVE);
+                compatibile.Add(BloodType.ZERO_NEGATIVE);
+            }
+            else if(bType == BloodType.A_POSITIVE)
+            {
+                compatibile.Add(BloodType.A_NEGATIVE);
+                compatibile.Add(BloodType.A_POSITIVE);
+                compatibile.Add(BloodType.ZERO_NEGATIVE);
+                compatibile.Add(BloodType.ZERO_POSITIVE);
+            }
+            else if(bType == BloodType.B_POSITIVE)
+            {
+                compatibile.Add(BloodType.B_NEGATIVE);
+                compatibile.Add(BloodType.B_POSITIVE);
+                compatibile.Add(BloodType.ZERO_NEGATIVE);
+                compatibile.Add(BloodType.ZERO_POSITIVE);
+            }
+            else if(bType == BloodType.B_NEGATIVE)
+            {
+                compatibile.Add(BloodType.B_NEGATIVE);
+                compatibile.Add(BloodType.ZERO_NEGATIVE);
+            }
+            else if(bType == BloodType.AB_POSITIVE)
+            {
+                compatibile.Add(BloodType.A_POSITIVE);
+                compatibile.Add(BloodType.B_POSITIVE);
+                compatibile.Add(BloodType.AB_POSITIVE);
+                compatibile.Add(BloodType.ZERO_POSITIVE);
+                compatibile.Add(BloodType.A_NEGATIVE);
+                compatibile.Add(BloodType.B_NEGATIVE);
+                compatibile.Add(BloodType.AB_NEGATIVE);
+                compatibile.Add(BloodType.ZERO_NEGATIVE);
+            }
+            else if(bType == BloodType.AB_NEGATIVE)
+            {
+                compatibile.Add(BloodType.A_NEGATIVE);
```

Comment:
Zero negative is in all conditions, so you can add it in front end leave it out of conditions